portal-frontend/src/app/features/applications/edit-submission/primary-contact/primary-contact.component.ts

```diff
@@ -275,4 +287,74 @@ export class PrimaryContactComponent extends FilesStepComponent implements OnIni
   private prepareGovernmentOwners() {
     this.parcelOwners = [];
   }
+
+  async onSelectPrimaryContactType(selectedThirdPartyAgent: boolean | null) {
+    if (this.form.dirty || (this.selectedOwnerUuid !== 'agent' && this.selectedOwnerUuid !== 'government')) {
+      await this.dialog
+        .open(PrimaryContactConfirmationDialogComponent, {
+          panelClass: 'no-padding',
+          disableClose: true,
+        })
+        .beforeClosed()
+        .subscribe(async (confirmed) => {
+          if (confirmed) {
+            this.switchPrimaryContactType(selectedThirdPartyAgent);
+          } else {
+            this.primaryContactType.setValue(selectedThirdPartyAgent);
+          }
+        });
+    } else {
+      this.switchPrimaryContactType(selectedThirdPartyAgent);
+    }
+  }
+
+  switchPrimaryContactType(selectedThirdPartyAgent: boolean | null) {
+    if (selectedThirdPartyAgent) {
+      this.onSelectAgent();
+    } else if (this.isGovernmentUser) {
+      this.onSelectGovernment();
+      if (this.parcelOwners.length == 1) {
+        this.onSelectOwner(this.parcelOwners[0].uuid);
+      }
```

Comment:
Use strict equality `===`. Also, you might want to check the logic here, because if a government user is signed in and the parcel has only one owner, it will run everything in this block, which is basically running `onSelectOwner` twice